src/torchcodec/decoders/_core/VideoDecoder.cpp

```diff
@@ -206,7 +206,8 @@ VideoDecoder::BatchDecodedOutput::BatchDecodedOutput(
 
 bool VideoDecoder::DecodedFrameContext::operator==(
     const VideoDecoder::DecodedFrameContext& other) {
-  return decodedWidth == other.decodedWidth && decodedHeight == decodedHeight &&
```

Comment:
Note that the stricter compilation revealed a real logic error here, as we had the code `decodedHeight == decodedHeight`.